### ISSUE TITLE
js2-mode in spacemacs 0.200.13: no syntax highlighting until saving file and reverting, strange autocompletion

### ISSUE THREAD

**Issue by dennismayr** [syl20bnr/spacemacs#11782]
Hello community,

Javascript syntax isn't highlighted in `js2-mode` until saving file and reverting to saved.
No configuration changes made, package defaults used.
![no_syntax_highlight_1](https://i.imgur.com/PefS1j4.png)

Same as https://github.com/mooz/js2-mode/issues/91, but still happening today, using Spacemacs (0.200.13, master branch).
I tried a fresh Emacs install, a clean Spacemacs install + default dotfile, and then added `js2-mode`.
Syntax is only highlighted after pressing 'enter' / 'return', and not on the fly as in other modes.

Expected: syntax highlighting on-the-fly
Experienced: no syntax highlighting until either pressing 'enter' / 'return' on a clean Spacemacs installation, or saving file and reverting to saved, in customized Spacemacs installation.

Sequence of events:
![saved_file](https://i.imgur.com/EYcfO8e.png)
![revert_file](https://i.imgur.com/7hCq6jK.png)
![syntax_highlight_ok](https://i.imgur.com/ATGPyvO.png)

In short, syntax highlighting only works after hitting 'enter' / 'return' on a Spacemacs fresh install with only `js2-mode` added, or after saving / reverting from saved on a Spacemacs installation with other additional packages.

However, `js2-mode` syntax highlighting *does work* on _plain Emacs_ 26.1, elementaryOS / Ubuntu 16.04.

**Comment by dennismayr** [comment]
Here's the case with cleam Spacemacs + `js2-mode`: code being typed is shown in red:
![code_red](https://i.imgur.com/VpqHeaA.png)

It is highlighted only after typing 'enter' / 'return':
![highlighted](https://i.imgur.com/ZG5PBHf.png)

I do have other _javascript_-related packages installed, such as `rjsx-mode` and `xref-js2`.

**Comment by dennismayr** [comment]
I just tried typing the same code under `js-mode`.

Not only is `javascript` code highlighted immediately as expected, but autocompletion works as expected too, e.g. when typing _'console.log'_, _'cons'_ is autocompleted as _'const'_ under `js2-mode` and as _'console'_ under `js-mode`, which is expected; _'log'_ is autocompleted as `log` but suggesting logarithmic notation, e.g. `log()°`.

I'm not sure that autocomplete is taking `javascript` elements and syntax into account.

![js-mode_ok](https://i.imgur.com/cOpWJOl.png)
